thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/detection/DetectionJobScheduler.java

```diff
@@ -264,4 +280,241 @@ private Long getIdFromJobKey(String jobKey) {
     String id = tokens[tokens.length - 1];
     return Long.valueOf(id);
   }
+
+  /**
+   * Sequentially performs anomaly detection for all the monitoring windows that are located between backfillStartTime
+   * and backfillEndTime. A lightweight job is performed right after each detection job and notified is set to false in
+   * order to silence the mail alerts.
+   *
+   * NOTE: We assume that the backfill window for the same function DOES NOT overlap. In other words, this function
+   * does not guarantees correctness of the detections result if it is invoked twice with the same parameters.
+   *
+   * @param functionId the id of the anomaly function, which has to be an active function
+   * @param backfillStartTime the start time for backfilling
+   * @param backfillEndTime the end time for backfilling
+   */
+  public void runBackfill(long functionId, DateTime backfillStartTime, DateTime backfillEndTime) {
+    AnomalyFunctionDTO anomalyFunction = anomalyFunctionDAO.findById(functionId);
+    boolean isActive = anomalyFunction.getIsActive();
+    if (!isActive) {
+      LOG.info("Skipping function {}", functionId);
+      return;
+    }
+
+    BackfillKey backfillKey = new BackfillKey(functionId, backfillStartTime, backfillEndTime);
+    Thread returnedThread = existingBackfillJobs.putIfAbsent(backfillKey, Thread.currentThread());
```

Comment:
should we not check the database if there is something already running? looks  like we are maintaining the list of currently running jobs in memory.